configuration/set.go

```diff
@@ -0,0 +1,84 @@
+package configuration
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	. "github.com/sonatype-nexus-community/nancy/logger"
+	"github.com/sonatype-nexus-community/nancy/types"
+	"gopkg.in/yaml.v2"
+)
+
+type IQConfig struct {
+	Server   string `yaml:"Server"`
+	Username string `yaml:"Username"`
+	Token    string `yaml:"Token"`
+}
+
+type OSSIndexConfig struct {
+	Username string `yaml:"Username"`
+	Token    string `yaml:"Token"`
+}
+
+var homeDir string
+
+func GetConfigFromCommandLine() {
+	homeDir, _ = os.UserHomeDir()
+	fmt.Print("Hi! What config can I help you set, IQ or OSS Index (values: iq, ossindex, enter for exit)? ")
+	var configType string
+	fmt.Scanln(&configType)
+	switch configType {
+	case "iq":
+		getAndSetIQConfig()
+	case "ossindex":
+		getAndSetOSSIndexConfig()
+	case "":
+		os.Exit(0)
+	default:
+		fmt.Println("Invalid value, iq and ossindex are accepted values, try again!")
+		GetConfigFromCommandLine()
+	}
+}
+
+func getAndSetIQConfig() {
+	iqConfig := IQConfig{Server: "http://localhost:8070", Username: "admin", Token: "admin123"}
+	fmt.Print("What is the address of your Nexus IQ Server (default: http://localhost:8070)? ")
+	fmt.Scanln(&iqConfig.Server)
+	fmt.Print("What username do you want to authenticate as (default: admin)? ")
+	fmt.Scanln(&iqConfig.Username)
+	fmt.Print("What token do you want to use (default: admin123)? ")
```

Comment:
Yeah, which is why I defaulted it! I could also just warn someone if they use the defaults and go "DO YOU REALLY WANT TO DO THIS?" and if they say no, rerun the function and recapture them? WDYT?

Reply:
Yeah I'm cool with that

Reply:
How is this?

```
(base) 504 jeffryhesse:nancy (ConfigOptions *)$ ./nancy config
Hi! What config can I help you set, IQ or OSS Index (values: iq, ossindex, enter for exit)? iq
What is the address of your Nexus IQ Server (default: http://localhost:8070)? 
What username do you want to authenticate as (default: admin)? 
What token do you want to use (default: admin123)? 

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!! WARNING : You are using the default username and password for Nexus IQ. !!!!
!!!! You are strongly encouraged to change these, and use a token.           !!!!
!!!! Would you like to change them and try again?                            !!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

[y/N]? y
What is the address of your Nexus IQ Server (default: http://localhost:8070)? 
What username do you want to authenticate as (default: admin)? 
What token do you want to use (default: admin123)? 

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!!! WARNING : You are using the default username and password for Nexus IQ. !!!!
!!!! You are strongly encouraged to change these, and use a token.           !!!!
!!!! Would you like to change them and try again?                            !!!!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

[y/N]? N
```

Reply:
I like the flowand overall it's good. Might tweak this 
`You are using the default username and password `

From an **and** to an **and/or** since you could change one and not the other.

Reply:
Did that for you and changed the check to a `||` as well